File: skills/E-personal-life/C18-life-health-physical-world/CaseMark/skills/_root-skill/_repository-root/skills/legal/authority-verification/scripts/verify_authority.ts
#!/usr/bin/env node

type Args = {
  citation: string;
  highlightQuery: string;
  relatedCount: number;
};

function parseArgs(argv: string[]): Args {
  let citation = "";
  let highlightQuery = "holding";
  let relatedCount = 5;

  for (let i = 0; i < argv.length; i += 1) {
    const arg = argv[i];
    if (arg === "--citation") {
      citation = argv[i + 1] ?? "";
      i += 1;
      continue;
    }
    if (arg === "--highlight-query") {
      highlightQuery = argv[i + 1] ?? highlightQuery;
      i += 1;
      continue;
    }
    if (arg === "--related-count") {
      relatedCount = Number.parseInt(argv[i + 1] ?? "5", 10);
      i += 1;
    }
  }

  if (!citation) {
    throw new Error("usage: npx tsx scripts/verify_authority.ts --citation '531 U.S. 98' [--highlight-query holding] [--related-count 5]");
  }

  if (!Number.isFinite(relatedCount) || relatedCount < 1) {
    throw new Error("--related-count must be a positive integer");
  }

  return { citation, highlightQuery, relatedCount };
}

async function main() {
  const apiKey = process.env.CASEDEV_API_KEY;
  if (!apiKey) {
    throw new Error("CASEDEV_API_KEY is required");
  }

  const args = parseArgs(process.argv.slice(2));
  let CasedevModule: typeof import("casedev");
  try {
    CasedevModule = await import("casedev");
  } catch {
    throw new Error("Missing dependency: install with 'npm install casedev' and run with 'npx tsx'");
  }

  const Casedev = CasedevModule.default;
  const client = new Casedev({ apiKey });
  const verification = await client.legal.verify({ text: args.citation });

  const results = [];

  for (const item of verification.citations) {
    const result: Record<string, unknown> = {
      original: item.original,
      status: item.status,
    };

    if (item.status === "verified") {
      result.case = {
        name: item.case.name,
        url: item.case.url,
      };

      const [fullText, similar] = await Promise.all([
        client.legal.fullText({
          url: item.case.url,
          highlightQuery: args.highlightQuery,
          maxCharacters: 12000,
        }),
        client.legal.similar({
          url: item.case.url,
          numResults: args.relatedCount,
        }),
      ]);

      result.highlights = fullText.highlights ?? [];
      result.relatedAuthorities = similar.similarSources.map((source) => ({
        title: source.title,
        url: source.url,
      }));
    }

    if (item.status === "multiple_matches") {
      result.candidates = (item.candidates ?? []).map((candidate) => ({
        name: candidate.name,
        url: candidate.url,
      }));
    }

    results.push(result);
  }

  const report = {
    citation: args.citation,
    summary: {
      total: verification.summary.total,
      verified: verification.summary.verified,
      notFound: verification.summary.notFound,
      multipleMatches: verification.summary.multipleMatches,
    },
    results,
  };

  console.log(JSON.stringify(report, null, 2));
}

main().catch((error) => {
  const message = error instanceof Error ? error.message : String(error);
  console.error(message);
  process.exit(1);
});

File: skills/E-personal-life/C18-life-health-physical-world/CaseMark/skills/_root-skill/_repository-root/skills/legal/authority-verification/scripts/verify_authority.py
#!/usr/bin/env python3
"""Minimal case.dev authority-verification example.

Verifies a citation, retrieves source text for verified matches,
and optionally expands to related authorities.
"""

from __future__ import annotations

import argparse
import json
import os
import sys
from typing import Any


def serialize_summary(summary: Any) -> dict[str, Any]:
    return {
        "total": getattr(summary, "total", None),
        "verified": getattr(summary, "verified", None),
        "not_found": getattr(summary, "not_found", None),
        "multiple_matches": getattr(summary, "multiple_matches", None),
    }


def build_parser() -> argparse.ArgumentParser:
    parser = argparse.ArgumentParser(description="Verify a legal citation with case.dev.")
    parser.add_argument("--citation", required=True, help="Citation text to verify, e.g. '531 U.S. 98'")
    parser.add_argument(
        "--highlight-query",
        default="holding",
        help="Highlight query for full-text retrieval when verification succeeds.",
    )
    parser.add_argument(
        "--related-count",
        type=int,
        default=5,
        help="Number of related authorities to request with legal.similar().",
    )
    return parser


def main() -> int:
    parser = build_parser()
    args = parser.parse_args()

    api_key = os.environ.get("CASEDEV_API_KEY")
    if not api_key:
        print("CASEDEV_API_KEY is required", file=sys.stderr)
        return 1

    try:
        import casedev
    except ImportError:
        print("Missing dependency: install with 'python3 -m pip install casedev'", file=sys.stderr)
        return 1

    client = casedev.Casedev(api_key=api_key)

    try:
        verification = client.legal.v1.verify(text=args.citation)
    except Exception as exc:
        print(f"case.dev verification failed: {exc}", file=sys.stderr)
        return 1

    report: dict[str, Any] = {
        "citation": args.citation,
        "summary": serialize_summary(getattr(verification, "summary", None)),
        "results": [],
    }

    for item in verification.citations:
        result: dict[str, Any] = {
            "original": item.original,
            "status": item.status,
        }

        if item.status == "verified":
            case = item.case
            result["case"] = {
                "name": case.name,
                "url": case.url,
            }

            try:
                full_text = client.legal.v1.get_full_text(
                    url=case.url,
                    highlight_query=args.highlight_query,
                    max_characters=12000,
                )
                similar = client.legal.v1.similar(
                    url=case.url,
                    num_results=args.related_count,
                )
            except Exception as exc:
                print(f"case.dev follow-on retrieval failed: {exc}", file=sys.stderr)
                return 1

            result["highlights"] = list(getattr(full_text, "highlights", []))
            result["related_authorities"] = [
                {"title": source.title, "url": source.url}
                for source in getattr(similar, "similar_sources", [])
            ]

        elif item.status == "multiple_matches":
            result["candidates"] = [
                {"name": candidate.name, "url": candidate.url}
                for candidate in getattr(item, "candidates", [])
            ]

        report["results"].append(result)

    print(json.dumps(report, indent=2, default=str))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
